### DISCUSSION TITLE
Homekit Passthrough

### DISCUSSION THREAD

**Discussion by Zehir** [deconz-community/node-red-contrib-deconz#194] [General]
See below

**Comment by Nazarbayevv** [comment]
Hello,

Please, can you explain, what's wrong in HomeKit’s “Messages Passthrough”? Why you added “ The msg that go through HomeKit and back to an output node are now discarded with a warning message.”? And how are you recommend to use switch + light + homekit now? 
![498098B7-EA6C-4CCF-BAAB-CE0600AEA69C](https://user-images.githubusercontent.com/50654913/149217000-4a43e7e0-e155-4cc9-8cd7-ba1f1bd198bd.jpeg)

Thanks a lot.

**Reply by Zehir** [reply]
> Hello,
> 
> Please, can you explain, what's wrong in HomeKit’s “Messages Passthrough”? Why you added “ The msg that go through HomeKit and back to an output node are now discarded with a warning message.”? And how are you recommend to use switch + light + homekit now? ![498098B7-EA6C-4CCF-BAAB-CE0600AEA69C](https://user-images.githubusercontent.com/50654913/149217000-4a43e7e0-e155-4cc9-8cd7-ba1f1bd198bd.jpeg)
> 
> Thanks a lot.

It's because this is causing a loop, A light update homekit and update the light back. For a loop like this you need do disable the Messages passthrough. For a switch you can just have an other path with switch -> light or you can define rules in the phoscon app

**Reply by Nazarbayevv** [reply]
Don you mean like this?
![D0FDC121-7346-4BF7-8F19-9EA235DB3DF9](https://user-images.githubusercontent.com/50654913/149230651-10cae83e-e907-43dd-ad00-05aed14b2240.jpeg)

**Reply by Zehir** [reply]
> Don you mean like this?
> ![D0FDC121-7346-4BF7-8F19-9EA235DB3DF9](https://user-images.githubusercontent.com/50654913/149230651-10cae83e-e907-43dd-ad00-05aed14b2240.jpeg)
> 

No just what you did before but just without the passtrough option

**Reply by TheNON75** [reply]
Hi,

If HomeKit pass through is enabled, all messages that are sent to the HomeKit node are "passing/going through" it and will appear on its output.

Deconz out node is to send commands to deconz 

Deconz in node is to receive updates from deconz

In this case:
- deconz "in" sends an updated state or whatever information to the HomeKit node to reflect the current situation with the device/accessory in the home app
- as pass through is enabled, this information will appear on the output of the HomeKit node and sent to deconz "out"
- the msg sent to deconz "out" will initiate an action in deconz and the results will appear on the deconz "in", which is sent to the HomeKit node and .... loop

In case you have doubts or questions on how to manage or avoid pass through in NRCHKB, please visit their GitHub or discord

**Comment by Nazarbayevv** [comment]
Thanks. I understood this moment.
